dynosaur/tests/pass/basic-mut.stdout

```diff
@@ -27,56 +27,40 @@ mod _dynosaur_macro_DynMyTrait {
             Box::pin(<Self as MyTrait>::foo(self))
         }
     }
+    #[repr(transparent)]
     pub struct DynMyTrait<'dynosaur_struct> {
-        ptr: *mut (dyn ErasedMyTrait + 'dynosaur_struct),
-        owned: bool,
+        ptr: dyn ErasedMyTrait + 'dynosaur_struct,
     }
     impl<'dynosaur_struct> MyTrait for DynMyTrait<'dynosaur_struct> {
         fn foo(&mut self) -> impl ::core::future::Future<Output = ()> {
             let fut:
                     ::core::pin::Pin<Box<dyn ::core::future::Future<Output =
-                    ()> + '_>> = unsafe { &mut *self.ptr }.foo();
+                    ()> + '_>> = self.ptr.foo();
             let fut:
                     ::core::pin::Pin<Box<dyn ::core::future::Future<Output =
                     ()> + 'static>> = unsafe { ::core::mem::transmute(fut) };
             fut
         }
     }
     impl<'dynosaur_struct> DynMyTrait<'dynosaur_struct> {
-        pub fn new<DYNOSAUR>(value: DYNOSAUR) -> Self where
-            DYNOSAUR: MyTrait + 'dynosaur_struct {
-            let value = Box::new(value);
-            Self {
-                ptr: Box::into_raw(value as
-                            Box<dyn ErasedMyTrait + 'dynosaur_struct>) as
-                    *mut (dyn ErasedMyTrait + 'dynosaur_struct),
-                owned: true,
-            }
+        pub fn new(value: Box<impl MyTrait>)
+            -> Box<DynMyTrait<'dynosaur_struct>> {
+            let value: Box<dyn ErasedMyTrait> = value;
+            unsafe { ::core::mem::transmute(value) }
         }
```

Comment:
The lifetime is unconstrained. We need to account for the lifetime of the `impl MyTrait` somehow.

The transmute should not change any lifetimes, only swap out `dyn ErasedMyTrait` with `DynMyTrait`.

```suggestion
        pub fn new<T: MyTrait + 'dynosaur_struct>(value: Box<T>)
            -> Box<DynMyTrait<'dynosaur_struct>> {
            let value: Box<dyn ErasedMyTrait + 'dynosaur_struct> = value;
            unsafe { ::core::mem::transmute(value) }
        }
```